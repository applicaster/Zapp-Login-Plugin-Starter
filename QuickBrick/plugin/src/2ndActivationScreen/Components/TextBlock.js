import React from 'react';
import {
  View,
  Text,
  ActivityIndicator
} from 'react-native';

function TextBlock(props) {
  const {
    goTo = '',
    activationUrl = '',
    codeInstructions = '',
    loading = false,
    customStyles = {},
    pinCode = ''
  } = props;

  return (
    <>
      <Text
        style={{ ...customStyles.goToTextStyle, ...styles.text }}
        numberOfLines={1}
        ellipsizeMode="tail"
      >
        {goTo}
      </Text>
      <Text
        style={{ ...customStyles.activationUrlStyle, ...styles.url }}
        adjustsFontSizeToFit
      >
        {activationUrl}
      </Text>
      <Text
        style={{ ...customStyles.codeInstructionsStyle, ...styles.url }}
        numberOfLines={1}
        ellipsizeMode="tail"
      >
        {codeInstructions}
      </Text>
      {
        loading
          ? (
            <View style={styles.pinCodeSpinner}>
              <ActivityIndicator size="small" color="#525A5C" />
            </View>
          )
          : (
            <Text
              style={customStyles.activationCodeStyle}
              adjustsFontSizeToFit
            >
              {pinCode}
            </Text>
          )
      }
    </>
  );
}

export default TextBlock;

const styles = {
  text: {
    marginBottom: 20
  },
  url: {
    marginBottom: 60
  },
  pinCodeSpinner: {
    width: 500,
    alignItems: 'center',
    justifyContent: 'center'
  }
};
