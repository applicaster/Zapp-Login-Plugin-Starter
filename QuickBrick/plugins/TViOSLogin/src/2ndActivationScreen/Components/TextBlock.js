import React from 'react';
import {
  View,
  Text,
  ActivityIndicator
} from 'react-native';

function TextBlock(props) {
  const {
    goTo = '',
    mainInstructions = '',
    activationUrl = '',
    codeInstructions = '',
    loading = false,
    customStyles = {},
    pinCode = ''
  } = props;

  return (
    <>
      <Text
        style={{ ...customStyles.mainTextStyle, ...styles.title }}
        numberOfLines={2}
        ellipsizeMode="tail"
      >
        {mainInstructions}
      </Text>
      <Text
        style={{ ...customStyles.goToTextStyle, ...styles.text }}
        numberOfLines={1}
        ellipsizeMode="tail"
      >
        {goTo}
      </Text>
      <Text
        style={{ ...customStyles.activationUrlStyle, ...styles.url }}
        adjustsFontSizeToFit
      >
        {activationUrl}
      </Text>
      <Text
        style={customStyles.codeInstructionsStyle}
        numberOfLines={1}
        ellipsizeMode="tail"
      >
        {codeInstructions}
      </Text>
      <View style={styles.activationCode}>
        {
          loading
            ? <ActivityIndicator size="small" color="#D8D8D8" />
            : (
              <Text
                style={customStyles.activationCodeStyle}
                adjustsFontSizeToFit
              >
                {pinCode}
              </Text>
            )
        }
      </View>
    </>
  );
}

export default TextBlock;

const styles = {
  title: {
    fontWeight: 'bold',
    marginTop: '1.4%',
    marginBottom: '8%'
  },
  text: {
    marginBottom: '1.4%'
  },
  url: {
    marginBottom: '2%',
    fontWeight: 'bold'
  },
  activationCode: {
    fontWeight: 'bold',
    marginTop: '5.2%',
    width: '80%'
  }
};
