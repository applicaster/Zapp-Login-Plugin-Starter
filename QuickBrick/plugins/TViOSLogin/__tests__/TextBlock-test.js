import React from 'react';
import renderer from 'react-test-renderer';
import TextBlock from '../src/2ndActivationScreen/Components/TextBlock';

describe('TextBlock', () => {
  it('renders correctly', () => {
    const defaultProps = {
      goTo: '',
      activationUrl: '',
      codeInstructions: '',
      loading: false,
      customStyles: {},
      pinCode: ''
    };
    const tree = renderer
      .create(<TextBlock props={defaultProps} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('check loading indicator displayed', () => {
    const props = {
      loading: true
    };

    const testRenderer = renderer.create(<TextBlock props={props} />);
    const testInstance = testRenderer.root;

    const spinner = testInstance.findAllByType('ActivityIndicator')[0];
    expect(spinner);
  });
});
