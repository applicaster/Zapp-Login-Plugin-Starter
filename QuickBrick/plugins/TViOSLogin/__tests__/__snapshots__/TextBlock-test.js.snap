// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBlock renders correctly 1`] = `
Array [
  <Text
    ellipsizeMode="tail"
    numberOfLines={2}
    style={
      Object {
        "fontWeight": "bold",
        "marginBottom": "8%",
        "marginTop": "1.4%",
      }
    }
  >
    
  </Text>,
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "marginBottom": "1.4%",
      }
    }
  >
    
  </Text>,
  <Text
    adjustsFontSizeToFit={true}
    style={
      Object {
        "fontWeight": "bold",
        "marginBottom": "2%",
      }
    }
  >
    
  </Text>,
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
  >
    
  </Text>,
  <View
    style={
      Object {
        "fontWeight": "bold",
        "marginTop": "5.2%",
        "width": "80%",
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
    >
      
    </Text>
  </View>,
]
`;
